### PR TITLE
Refactor CNI documentation.

### PR DESCRIPTION
**Pull request by philwinder**
- Add Mesos CNI instructions
- Rename page to something more generic.
- Fix old links.
- Minor editing.

Review: @abuehrle 
cc: @bboreham, @pidster

### CODE REVIEW

**Comment by abuehrle** [comment]
LGTM although, I do have some other edits on that page that I can address on a different PR.  Also, we should have a guide that shows an example of this as well, which I'll mention to @lukemarsden

File: site/cni-plugin.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 ---
-title: Integrating Kubernetes via the CNI Plugin
+title: Weave CNI Plugin
```
Comment (rade) [inline site/cni-plugin.md]: @abuehrle I thought we wanted all headings to be "actions". Also, I am concerned that w/o k8s/mesos being mentioned in the heading, folks really won't know know what this is about - 'CNI' is not a well-known term.

So I'd opt for 'Integrating Kubernetes and Mesos via the CNI Plugin', but that's rather long.

Comment (abuehrle) [inline site/cni-plugin.md]: How about:  "Configuring the CNI Plugin for Kubernetes and Mesos"

Comment (rade) [inline site/cni-plugin.md]: IMO that's putting the cart before the horse. "Configuring the CNI Plugin" is not something a user wants to do. It's a means to and end. And the end is using Weave Net with K8s or Mesos.

Also, note the two TOC entries just above this one:
- Integrating Docker via the API Proxy
- Integrating Docker via the Network Plugin

The title I suggested follows the same pattern.

Comment (abuehrle) [inline site/cni-plugin.md]: sure.

Comment (abuehrle) [inline site/cni-plugin.md]: 'Integrating Kubernetes and Mesos via the CNI Plugin' is fine with me, if you'd like to keep the same structure.  Regarding the action rule.  It's not an all in rule, IMO.  Normally you'd want the headings to be action-orientated if the user is required to do something.  But if it's straight up description or an explanation, then I'd say a regular title is just fine.  Sometimes a document might have a combination of these titles.

Comment (rade) [inline site/cni-plugin.md]: @philwinder would you mind making that small change?

**Comment by philwinder** [comment]
@rade: Done.

**Pull request merged**